### PR TITLE
doc: Update Unit Test Readme

### PR DESCRIPTION
**Pull request by RoboticMind**
Switches the unit test README to use markdown (displays better on GitHub and has better links), changes a dead link, and simplifies some writing

### CODE REVIEW

File: src/test/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+convention is "<source_filename>_tests.cpp" and such files should wrap
+their tests in a test suite called "<source_filename>_tests". For an
+examples of this pattern, examine uint160_tests.cpp and uint256_tests.cpp.
+
+The tests in tranaction_tests.cpp are edge cases of bitcoin transactions.
+They are in their current state not relevant for gridcoin. Unusual transactions
+should be collected again from the gridcoin blockchain and replace
+the current test cases.
+
+For further reading, [see Boost's documenation](https://www.boost.org/doc/libs/1_73_0/libs/test/doc/html/boost_test/intro.html)
```
Comment (div72) [inline src/test/README.md]:
```suggestion
For further reading, [see Boost's documentation](https://www.boost.org/doc/libs/1_73_0/libs/test/doc/html/boost_test/intro.html)
```

File: src/test/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+that runs all of the unit tests. The main source file is called
+test_gridcoin.cpp, which simply includes other files that contain the
+actual unit tests (outside of a couple required preprocessor
+directives). The pattern is to create one test file for each class or
+source file for which you want to create unit tests. The file naming
+convention is "<source_filename>_tests.cpp" and such files should wrap
+their tests in a test suite called "<source_filename>_tests". For an
+examples of this pattern, examine uint160_tests.cpp and uint256_tests.cpp.
+
+The tests in tranaction_tests.cpp are edge cases of bitcoin transactions.
```
Comment (div72) [inline src/test/README.md]:
```suggestion
The tests in transaction_tests.cpp are edge cases of Gridcoin transactions.
```

File: src/test/README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+test_gridcoin.cpp, which simply includes other files that contain the
+actual unit tests (outside of a couple required preprocessor
+directives). The pattern is to create one test file for each class or
+source file for which you want to create unit tests. The file naming
+convention is "<source_filename>_tests.cpp" and such files should wrap
+their tests in a test suite called "<source_filename>_tests". For an
+examples of this pattern, examine uint160_tests.cpp and uint256_tests.cpp.
+
+The tests in transaction_tests.cpp are edge cases of Gridcoin transactions.
+They are in their current state not relevant for gridcoin. Unusual transactions
```
Comment (cyrossignol) [inline src/test/README.md]:
```suggestion
They are in their current state not relevant for Gridcoin. Unusual transactions
```
7835-48x48x32

**Review by cyrossignol — APPROVED** [review]
